### PR TITLE
feat(dsmiddleware/splitop): add splitop middleware

### PR DESCRIPTION
**Pull request by vvakame**
Please read the CLA carefully before submitting your contribution to Mercari.
Under any circumstances, by submitting your contribution, you are deemed to accept and agree to be bound by the terms and conditions of the CLA.

https://www.mercari.com/cla/

### CODE REVIEW

File: dsmiddleware/splitcall/splitcall.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+func (sh *splitHandler) PutMultiWithoutTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) ([]datastore.Key, error) {
+	return info.Next.PutMultiWithoutTx(info, keys, psList)
+}
+
+func (sh *splitHandler) PutMultiWithTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) ([]datastore.PendingKey, error) {
+	return info.Next.PutMultiWithTx(info, keys, psList)
+}
+
+func (sh *splitHandler) GetMultiWithoutTx(info *datastore.MiddlewareInfo, keys []datastore.Key, psList []datastore.PropertyList) error {
+	sh.logf(info.Context, "process %d keys", len(keys))
```
Comment (sinmetal) [inline dsmiddleware/splitcall/splitcall.go]: Is this log output permanently?

File: dsmiddleware/splitcall/splitcall.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+	var mErr datastore.MultiError = make([]error, len(keys))
+	var foundErr bool
+	next := info.Next
+	for i := 0; i < len(keys); i += sh.splitThreshold {
+		end := i + sh.splitThreshold
+		if len(keys) < end {
+			end = len(keys)
+		}
+		sh.logf(info.Context, "process [%d, %d) range keys", i, end)
```
Comment (sinmetal) [inline dsmiddleware/splitcall/splitcall.go]: Is this log output permanently?

File: dsmiddleware/splitcall/splitcall.go (hunk 1 above)
Comment (vvakame) [reply dsmiddleware/splitcall/splitcall.go]: yes. default implementation is `func(ctx context.Context, format string, args ...interface{}) {}`

File: dsmiddleware/splitcall/splitcall.go (hunk 2 above)
Comment (vvakame) [reply dsmiddleware/splitcall/splitcall.go]: same as above

**Comment by vvakame** [comment]
sinmetalが `thousandbreaker` にしましょう！！！って言ってきたけど却下した

**Pull request merged**